### ISSUE TITLE
Add ccache in the builds

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#174]
This is a spinoff of vectorisation issue #71 and a followup to the big PR #171.

With vectorization, I often systematically do several builds (none, sse4, avx2, 512y, 512z - times float, double - times cxtyperef or nocxtyperef), which starts to take quite some time, and often is a repetition of previous builds.

Adding support for ccache would speed this up enormously. Of course this woukld be easier with cmake, but the makefiles for vectorisation have quite some logic, I would not move to cmake yet.

To be investigated (I had all this set up nicely and smoothly for COOL in the past, https://gitlab.cern.ch/lcgcool/cool/-/blob/master/cmake/CORALEnableCCache.cmake).

**Comment by valassi** [comment]
I have fixed this with a simple PR #181.

To enable ccache, one has to set USECCACHE=1. It would be nicer to actually avoid all these patches and simply prepend ccache to $CXX and $NVCC externally, however there is some logic for NVCC in the Maekfile that depends on CUDA_HOME. So I just added this patch.

Note that ccache 3.7.7 (default on centos7) does not properly cache cuda builds where the '-x cu' option is used. This is fixed in the latest ccache 4.2.1. So I am using a private 4.2.1 build for my work.

I have also asked the SPI team to install the latest ccache on cvmfs, https://sft.its.cern.ch/jira/projects/SPI/issues/SPI-1878

The builds are now MUCH faster when already found in the cache.
